### PR TITLE
Theme modernization

### PR DESCRIPTION
**Pull request by AlessioGr**
I'll try to make the website look more modern. Currently implemented:

- [x] Make light the default theme. It's a lot more friendly and looks more modern. Come and fight me
- [x] If the visitor has a dark theme preference set, dark theme will be chosen as their default theme
- [x] Add theme switcher
- [x] Make the theme switcher remember your choice after reloads
- [x] Remove the ugly shadow from the header
- [x] Custom scrollbar
- [x] Replace the ugly gray backgrounds with some more friendly blue-ish gray
- [x] Fix the missing un-themed black parts in the light theme
- [x] Replace the filtered logo with a real, light-theme logo
- [x] Make the header responsive for mobile users

~~How the light theme currently looks:~~
image

How the light theme currently looks after getting electronicboy came. It should look more like the new papermc forums:
image

Without the announcement bar:
image

Comparison: PaperMC forums header with fixed logo and Hangar header:
image
image
Definitely more similar, right? The menu on the left would need to be added to the Paper forums. The Hangar buttons on the right (like the profile icons) can stay like that, in my opinion because the profile stuff is completely separate in Hangar anyways. The navbar text links can use some minor styling to replicate the blue line under the text when hovering over it, as well as the little arrows next to it. I might add that in a future smaller, separate PR.

On Mobile:
image

And the black theme:
image

Oh, and the navbar is also sticky now

### CODE REVIEW

File: frontend/components/layouts/Header.vue [hunk 1]
```diff
@@ -15,6 +16,26 @@
...
         <v-spacer />
 
+        <div>
+            <v-tooltip v-if="!$vuetify.theme.dark" bottom>
+                <template #activator="{ on }">
+                    <v-btn color="info" class="mr-1" v-on="on" @click="darkMode">
+                        <v-icon>mdi-weather-night</v-icon>
+                    </v-btn>
+                </template>
+                <span>Turn on Dark Mode</span>
```
Comment (MiniDigger) [inline frontend/components/layouts/Header.vue:26]: needs to be translated

File: frontend/components/layouts/Header.vue [hunk 2]
```diff
@@ -15,6 +16,26 @@
...
+            </v-tooltip>
+
+            <v-tooltip v-else bottom>
+                <template #activator="{ on }">
+                    <v-btn color="info" class="mr-1" v-on="on" @click="darkMode">
+                        <v-icon color="yellow">mdi-white-balance-sunny</v-icon>
+                    </v-btn>
+                </template>
+                <span>Turn off Dark Mode</span>
```
Comment (MiniDigger) [inline frontend/components/layouts/Header.vue:35]: i18n

File: frontend/components/layouts/Header.vue (hunk 1 above)
Comment (AlessioGr) [reply frontend/components/layouts/Header.vue]: Done!

File: frontend/components/layouts/Header.vue (hunk 2 above)
Comment (AlessioGr) [reply frontend/components/layouts/Header.vue]: Done!

**Comment by AlessioGr** [comment]
I pretty much made most changes I wanted to make, especially in terms of the light mode & making Hangar look more like the Paper forums.

I think this should be merged, so further, smaller changes which improve some details can be made in smaller PRs. It's for sure not perfect at this state, but that way other people can base their changes pr these changes.

**Pull request merged**